src/rule_definition_tools.jl

```diff
@@ -79,69 +161,84 @@ macro scalar_rule(call, maybe_setup, partials...)
             call.args[i] = esc(arg)
         end
     end
-    if all(Meta.isexpr(partial, :tuple) for partial in partials)
-        input_rep = :(first(promote($(inputs...))))  # stand-in with the right type for an input
-        forward_rules = Any[rule_from_partials(input_rep, partial.args...) for partial in partials]
-        reverse_rules = Any[]
-        for i in 1:length(inputs)
-            reverse_partials = [partial.args[i] for partial in partials]
-            push!(reverse_rules, rule_from_partials(inputs[i], reverse_partials...))
+
+    # For consistency in code that follows we make all partials tuple expressions
+    partials = map(partials) do partial
+        if Meta.isexpr(partial, :tuple)
+            partial
+        else
+            @assert length(inputs) == 1
```

Comment:
is there anyway this could ever fail? I mean, i presume not, otherwise you wouldn't make it an assert... but just double checking!

Reply:
It actually could fail, 
I should change it it was an assertion before this PR and I just moved it,
but I might was well fix it on the way past.

```suggestion
            length(inputs) == 1 || error("Invalid use of `@scalar_rule`")
```

I'm OK with using `error` here as it can't be recovered from and is a programemer mistake